builtin/providers/aws/resource_aws_cloudfront_distribution.go

```diff
@@ -530,6 +540,19 @@ func resourceAwsCloudFrontDistributionRead(d *schema.ResourceData, meta interfac
 	d.Set("last_modified_time", aws.String(resp.Distribution.LastModifiedTime.String()))
 	d.Set("in_progress_validation_batches", resp.Distribution.InProgressInvalidationBatches)
 	d.Set("etag", resp.ETag)
+	d.Set("arn", resp.Distribution.ARN)
+
+	cloudFrontArn := resp.Distribution.ARN
+	tagResp, tagErr := conn.ListTagsForResource(&cloudfront.ListTagsForResourceInput{
+		Resource: cloudFrontArn,
+	})
+
+	if tagErr != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", cloudFrontArn)
+	}
+
+	d.Set("tags", tagsToMapCloudFront(tagResp.Tags))
```

Comment:
if `tagErr` is non-nil, seems like there would be a decent chance that `tagResp` is nil, does that sound right? We may want to guard here for a nil response and nil pointer dereference